documentation/ssr-and-hydration.mld

```diff
@@ -0,0 +1,301 @@
+{0 SSR and hydration}
+
+{1 How does React handle SSR markup?}
+During server-side rendering (SSR), server-reason-react generates the static HTML markup on the server. When this markup reaches the client's browser, React.js performs a process called "hydration", 
+it attaches event handlers and other interactive behaviors to the existing HTML elements, rather than re-rendering the entire page from scratc
+
+For hydration to work correctly, the initial server markup must match exactly what React expects on the client. 
+If there is a mismatch, React will throw a hydration error.
+
+For example:
+- server renders:[<span>Hello from server</span>]
+- client renders:[<span>Hello from client</span>]
+
+With this mismatch, instead of being able to efficiently "hydrate" the existing server markup with interactivity, React will throw away the server-rendered content and do a full client-side render, resulting in worse performance and user experience, like layout shifts and other annoyances.
```

Comment:
```suggestion
With this mismatch, instead of being able to efficiently hydrate the existing server markup with interactivity, React will throw away the server-rendered content and do a full client-side render, resulting in worse performance and user experience, like layout shifts and other annoyances.
```

You've used hydration/hydrate before, it's a defined technical jargon term now, you should drop the quotes.